### ISSUE TITLE
define fields

### ISSUE THREAD

**Issue by matthewberryman** [MSFREACH/msf-reach#179]
* What are mandatory and optional ones for events, reports and contacts.
* Types for fields (numbers, strings, etc.)
* Defining look and feel, default values.

**Comment by matthewberryman** [comment]
@anexact @LucieGueuning please refer to comments and replies I've put in the google sheets doc as some of the contacts fields need further clarifications. I'll take a look at fields for other parts of the system in coming days.
@talltom will be assisting review some proposed database schema changes I made in https://github.com/MSFREACH/msf-reach-schema/pull/25 and there's some corresponding work we'll need to do in the API in https://github.com/MSFREACH/msf-reach/issues/219 to reflect those changes for the date fields, then when I roll that out I will also need to update the client side code.
@mehrdadgit can you please assist with implementation of the updated & new contacts fields for now (see ^^ PR for google sheets link)

**Comment by LucieGueuning** [comment]
I’ve gone through the fields per (link to spreadsheet)
and added some further comments and replies—please check for these by looking for tabs with comments and then the cells with yellow triangles in the top right.

Some notable things: 
* on map layers, where are we at with regards to discussion around having a single page? There are some things that are harder / impossible to do (like updating the list of events on new events creation) without a single page.
* in the social media replies, we cannot for legal / privacy reasons and UX reasons link to event page.
* I’ve listed all the type and category response as requested to illustrate there’s overlap here, and there are UX reasons (due to mutual exclusivity and space on the forms) why I think only one question is preferred here.

I’ll get stuck back in to coding for the rest of the day and (since I had a late start on this today) also tomorrow as well, then resuming on Monday.

**Comment by matthewberryman** [comment]
@LucieGueuning I've put the MSF fields in per your request, now. I am still waiting on your feedback on type & category. It would be best from a UI perspective on the contact page (one less thing to fill out, and a *lot* less work in the code elsewhere especially display code, my import scripts) if we could just keep it to one expanded category. Or we could consider having the role like "supplier" filled in in the profile/speciality field?

**Comment by LucieGueuning** [comment]
working on that.

**Comment by LucieGueuning** [comment]
ok solved. 

So, **type of contact** needs to have a drop-down (as below) after 'personal details' information slide. 
**Category** needs to have a drop-down (as below) as part of a new slide (after contact details) for company information. 

![image](https://user-images.githubusercontent.com/25951555/35387304-0a79b3fc-01c7-11e8-8188-08754f27fd59.png)

**Comment by LucieGueuning** [comment]
@matthewberryman  all clear?

**Comment by matthewberryman** [comment]
No. You're asking essentially the same question twice—there's 85% overlap between "category" and "type" and these two words are synonyms so I'm not sure what the distinction is here especially with the significant overlap in answers. You'd mentioned the need to distinguish out suppliers as that's not necessarily mutually exclusive with NGO, however a lot of the other categories/types *are* mutually exclusive. Why not just have one list with "MSF Peer", "Current MSF staff", "Association member", "United Nations", "Government (excl. supplier)", "Other NGO (excl. supplier)", Academic, Private (medical?) practice, Other?
Then you have one meaningful field with no overlap, no need to ask people to fill in the same question twice (which is frustrating and time consuming), and this avoids a lot of work in the contact input and display UI, and also makes it less confusing in the display (as an operator I would be asking why have two fields with synonymous names that have the same answer most of the time?)

**Comment by matthewberryman** [comment]
So per discussion on Skype, we'll have a combined "type of contact" dropdown, then tick boxes for "MSF Peer" and "Supplier", and then under an employment section on same page make sure employer, job title, and division/unit are always options.

**Comment by matthewberryman** [comment]
Need updated_at field for events (also helps with part of #235)

**Comment by matthewberryman** [comment]
@LucieGueuning I had a bit of a look at the events UI to see what we're missing. The main things is around MSF Resources (people involved) - there is an option to list multiple people although only one space for nationality, and it's all freeform so prone to error. The easiest way to solve this I think is to link contacts to events https://github.com/MSFREACH/msf-reach/issues/40 although I have a question for you there, that I will ask soon, before I can proceed with that.

Separately on the edit page I need to update for changes to the severity slider. Anything else I notice I will post here although I'm pretty busy today so will be catching up a bit tonight and tomorrow.

**Comment by LucieGueuning** [comment]
MSF resource means staff working on the project itself and they are not part of the Contact database.

**Comment by matthewberryman** [comment]
Nationality -> visa required mapping needs to support multiple countries (at the moment only saves for one).

**Comment by matthewberryman** [comment]
various grammar & spelling / typo fixes required in the events edit modal on landing page

**Comment by matthewberryman** [comment]
Events need to have a description field. https://github.com/MSFREACH/msf-reach/issues/362

**Comment by matthewberryman** [comment]
The initial work was done, there are other issues such as #563 that cover further changes requested.
